homeassistant/components/wake_on_lan/switch.py

```diff
@@ -23,12 +23,13 @@
 
 DEFAULT_NAME = "Wake on LAN"
 DEFAULT_PING_TIMEOUT = 1
+DEFAULT_BROADCAST_PORT = 9
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_MAC): cv.string,
         vol.Optional(CONF_BROADCAST_ADDRESS): cv.string,
```

Comment:
The address also requires a default. For both, please use the `DEFAULT_` parameters in the `wakeonlan` package. https://github.com/remcohaszing/pywakeonlan/blob/master/wakeonlan.py

Reply:
Alternative is to not pass the args at all to the wlan function.

```python
kwargs = {}
if CONF_BROADCAST_PORT in call.data:
    kwargs['port'] = call.data[CONF_BROADCASTPORT]

await hass.async_add_job(
    partial(
        wakeonlan.send_magic_packet,
        mac_address,
        **kwargs,
    )
)
```

btw this is also broken in the `__init__.py` service.

Reply:
@balloob I have had a PR to fix this as well that sort of takes this approach. https://github.com/home-assistant/core/pull/37387

looks like there might be duplicate issues for the port option addition. I would not default the port in HA but leave that up to the library at this time.